x-pack/platform/plugins/shared/fleet/public/components/namespace_combo_box.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiFormRow, EuiComboBox, EuiLink, EuiIcon } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+
+interface NamespaceFormRowProps {
+  namespace?: string;
+  placeholder?: string;
+  isEditPage: boolean;
+  packageType?: string;
+  validationError?: string[] | null;
+  docLinks?: {
+    links: {
+      fleet: {
+        datastreamsNamingScheme: string;
+      };
+    };
+  };
+  onNamespaceChange: (namespace: string) => void;
+  'data-test-subj'?: string;
+  fullWidth?: boolean;
+  labelId?: string;
+  helpTextId?: string;
+}
+
+export const NamespaceComboBox: React.FC<NamespaceFormRowProps> = ({
+  namespace,
+  placeholder,
+  isEditPage,
+  packageType,
+  validationError,
+  docLinks,
+  onNamespaceChange,
+  'data-test-subj': dataTestSubj = 'packagePolicyNamespaceInput',
+  fullWidth = false,
+  labelId = 'xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyNamespaceInputLabel',
+  helpTextId = 'xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyNamespaceHelpLabel',
```

Comment:
NIt: Unless the `defaultMessage` for the help and label could be customized, I don't think we need these options